### ISSUE TITLE
Duplicate `npm-check-updates` in `.projen/deps.json`

### ISSUE THREAD

**Issue by polothy** [projen/projen#804]
While upgrading to `projen` to `0.17.95`, we noticed this duplicate block in`.projen/deps.json`:

```json
    {
       "name": "npm-check-updates",
       "version": "^11",
       "type": "build"
     },
     {
       "name": "npm-check-updates",
       "version": "^11",
       "type": "build"
     },
```

No idea if this is even a bad thing.  Seems like this line is somehow getting called twice on our project: https://github.com/projen/projen/blob/06465af316f9f87dd250105ce2ef49f6b3ff6ef5/src/upgrade-dependencies.ts#L95

**Trimmed down** version of our `.projenrc.js`:

```js
const { AwsCdkConstructLibrary, ProjectType } = require('projen');

const project = new AwsCdkConstructLibrary({
  buildWorkflow: false, /* Define a GitHub workflow for building PRs. */
  defaultReleaseBranch: 'main', /* The name of the main release branch. */
  dependabot: false, /* Include dependabot configuration. */
  mergify: false, /* Adds mergify configuration. */
  pullRequestTemplate: false, /* Include a GitHub pull request template. */
  releaseWorkflow: false, /* Define a GitHub workflow for releasing from "master" when new versions are bumped. */
  projectType: ProjectType.LIB, /* Which type of project this is (library/app). */
});

project.synth();
```

Cheers and thanks for any help in advance! 🍻

**Comment by Hunter-Thompson** [comment]
Can reproduce this issue with the following commands :

```
yarn add projen && yarn projen new node && cat .projen/deps.json | grep npm-check-updates
 ```

**Comment by polothy** [comment]
Found what triggers the doubling, in `.projenrc.js`:

```js
  depsUpgrade: DependenciesUpgradeMechanism.githubWorkflow({
    ignoreProjen: true,
  }),
```

If you flip `ignoreProjen` to `false` then the doubling up goes away.

**Comment by eladb** [comment]
We just need to dedup dependencies in the Dependencies system

**Comment by Chriscbr** [comment]
@eladb Fixed by #862
